### ISSUE TITLE
Unexpected error when use `Optional` type inside `if x != None` condition

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#391]
**Describe the bug**
I'm not sure if this is a by-design behavior.

Please see this example:
![image](https://user-images.githubusercontent.com/3479453/70120684-1f14e000-16a8-11ea-8f8b-6039f63c9019.png)

**To Reproduce**

Code:
```python
def fn(a:str):
    pass

def create_s() -> Optional[str]:
    pass

s = create_s() # type of s is Optional[str] here
if s:
    fn(s) #no error here

if s is not None:
    fn(s) #no error here
    
if s != None:
    fn(s) #error here
```

**Expected behavior**
I expect `if s != None` having the same affect with `if s is not None`, making type of `s` become `str` inside the if block.

**VS Code extension or command-line**
I'm using VS Code extension 1.1.7

**Additional context**
This issue can be easily avoid by changing `!=` to `is not`, I'm reporting this only because I think it might be a bug.

Thanks!

**Comment by erictraut** [comment]
Thanks for the request. You're correct in assuming that pyright currently handles "is None" and "is not None", but it doesn't handle "== None" and "!= None". Those are easy to add, as they're effectively the same operation. This will be in the next release.

**Comment by erictraut** [comment]
This is now fixed in 1.1.8, which I just published.
